gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/ContentBloomFilter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.gc;
+
+import com.google.common.hash.BloomFilter;
+import com.google.common.hash.Funnels;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.DeltaLakeTable;
+import org.projectnessie.model.IcebergTable;
+
+/** A utility class wrapping bloom filter functionality. */
+public class ContentBloomFilter implements Serializable {
+
+  private static final int SIZE = 10_000_000;
+  private static final long serialVersionUID = -693336833916979221L;
+
+  private final BloomFilter<Long> icebergContentFilter;
+  private final BloomFilter<String> deltaLakeContentFilter;
```

Comment:
Instead of big binary bloom filter, I am keeping content specific bloom filters (int type for views, long type for IcebergTable, string type for deltalake)

Reply:
I get that. But why do you need two at all?